### PR TITLE
Adds guidance about error boundaries to React.lazy docs

### LINKED ISSUE #915: Include <ErrorBoundary> info in React.lazy() docs
CC: @housseindjirdeh 

The https://web.dev/code-splitting-suspense/ docs do a great job of covering `React.lazy()` and `<Suspense>`. One thing that they don't mention, though, is that lazy-loading might fail (see [this presentation/site](https://pawll.glitch.me/) for some common failure scenarios), and that an [`<ErrorBoundary>`](https://reactjs.org/docs/error-boundaries.html) can be used to detect and handle that failure gracefully.

I am happy to submit a PR with some recommendations and links if you think that would be useful.

### PR DESCRIPTION
**Pull request by jeffposnick**
Fixes #915

Changes proposed in this pull request:

- Adds a new section that describes using an error boundary to handle lazy-loading failures.

---

If this PR adds **new content** please fill out the form below.
If this PR changes existing content you can delete the form.

---

### CODE REVIEW

File: src/site/content/en/react/code-splitting-suspense/index.md [hunk 1]
```diff
@@ -173,12 +173,70 @@ loading indicator. This can make the user experience feel more jarring.
...
  more features that would extend this even further. The
  [React 16.x roadmap](https://reactjs.org/blog/2018/11/27/react-16-roadmap.html)
  explains this in more detail.
 {% endAside %}
 
+## Handle loading failures
+
+`Suspense` allows you to display a temporary loading state while network
+requests are made under the hood. But what if those network requests fail for
+some reason? You might offline, or perhaps your web app is attempting to
```
Comment (housseindjirdeh) [inline src/site/content/en/react/code-splitting-suspense/index.md]:
```suggestion
some reason? You might be offline, or perhaps your web app is attempting to
```

File: src/site/content/en/react/code-splitting-suspense/index.md [hunk 2]
```diff
@@ -173,12 +173,70 @@ loading indicator. This can make the user experience feel more jarring.
...
+lazy-load a [versioned URL](/http-cache/#long-lived-caching-for-versioned-urls)
+that is out of date, and no longer available following a server redeployment.
+
+React has a standard pattern for gracefully handling these types of loading
+failures: using an error boundary. As described [in the documentation](https://reactjs.org/docs/error-boundaries.html),
+any React component can serve as an error boundary if it implements either (or
+both) of the lifecycle methods `static getDerivedStateFromError()` or
+`componentDidCatch()`.
+
+To detect and handle lazy-loading failures, you can wrap you `Suspense`
```
Comment (housseindjirdeh) [inline src/site/content/en/react/code-splitting-suspense/index.md]:
```suggestion
To detect and handle lazy-loading failures, you can wrap your `Suspense`
```

**Comment by jeffposnick** [comment]
Feedback addressed!

I think @robdodson still needs to approve/merge?

**Comment by housseindjirdeh** [comment]
Yep, needs a codeowner approval

@robdodson I know we were experimenting with having multiple codeowners before. Will it be easier if I was also a codeowner for the `react/` directory?

**Comment by robdodson** [comment]
> @robdodson I know we were experimenting with having multiple codeowners before. Will it be easier if I was also a codeowner for the react/ directory?

Sure thing, I'll put together a PR to add you.

**Pull request merged**